### PR TITLE
feautre/h5p-short-answer-list-implementation

### PR DESCRIPTION
**Pull request by rider-griffin**
- initial commit for implementation of h5p-short-answer-list
- implements saving and restoration of previous state
- implements emission of xapi completion event

### CODE REVIEW

**Review by wbk — APPROVED** [review]

File: short-answer-list.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  };
+
+  ShortAnswerList.prototype.createSubmissionButton = function () {
+    var self = this;
+
+    var $footerContainer = $("<div>", {
+      class: "h5p-short-answer-footer",
+    }).appendTo(self.$inner);
+
+    var $submitButton = $("<div>", {
```
Comment (zachrip) [inline short-answer-list.js]: Make buttons buttons, and submit buttons should have the correct button type "submit" as well: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type

File: short-answer-list.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+      });
+    } else {
+      self.$helpButton.remove();
+    }
+  };
+
+  ShortAnswerList.prototype.createSubmissionButton = function () {
+    var self = this;
+
+    var $footerContainer = $("<div>", {
```
Comment (zachrip) [inline short-answer-list.js:131]: Does a <footer> make sense here?

File: short-answer-list.js [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        score += elementInstance.getScore();
+      }
+    });
+
+    return score;
+  };
+
+  ShortAnswerList.prototype.getMaxScore = function () {
+    let maxScore = 0;
+    this.pageInstances.forEach(function (elementInstance) {
```
Comment (zachrip) [inline short-answer-list.js]: I think a filter with .length could improve readability/intent here

File: short-answer-list.js [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+
+    return inputArray;
+  };
+
+  /**
+   * Returns True if all required inputs are filled.
+   * @returns {boolean} True if all required inputs are filled.
+   */
+  ShortAnswerList.prototype.requiredInputsIsFilled = function () {
+    var requiredInputsIsFilled = true;
```
Comment (zachrip) [inline short-answer-list.js]: Don't use var

File: short-answer-list.js [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    });
+
+    return requiredInputsIsFilled;
+  };
+
+  /**
+   * Mark required input fields.
+   */
+  ShortAnswerList.prototype.markRequiredInputFields = function () {
+    this.pageInstances.forEach(function (elementInstance) {
```
Comment (zachrip) [inline short-answer-list.js]: In general I think we prefer native for of syntax to forEach

**Pull request merged**